src/hvmc_net/mutual_recursion.message

```diff
@@ -0,0 +1,18 @@
+Seems like you're trying to run some recursive function(s) on strict-mode:
+{refs}
+
+Due to the ultra-greedy nature of strict-mode, that might result in infinite loops.
+If the float-combinators optimization is not on, we recommend activating it.
+
+You have 2 options:
+
+1. Easy Fix: use lazy-mode.
+
+Just append the `-L` option to `HVM-Lang`, and it will run in lazy-mode. It has the advantage of not doing wasteful work, having an improved complexity class, and being compatible with unrestricted recursion! It has a small overhead though, and isn't compatible with GPU.
+
+2. Hard Fix: refactor the program to use lazy references.
+
+When a function reference is in head position of an application or is duplicated by the non-linear use of a `let` expression it will be greedly expanded, leading to an infinite loop.
+If the application is used written as a combinator, it will automatically lifted to a lazy reference, which usually breaks the recursion cycle. Alternatively, by using the built-in `data` and `match` syntax, hvm-lang will try to do this automatically.
```

Comment:
Maybe we should put a very small example here? Something like
> (e.g. If pattern matching on scott-encoded ADTs, write `@a @x(x @a (Foo a) a)` instead of `@a @x(x (Foo a))`